src/vs/workbench/contrib/terminalContrib/suggest/browser/lspTerminalModelContentProvider.ts

```diff
@@ -0,0 +1,183 @@
+		if (content !== 'exit()' && this._shellType === GeneralShellType.Python) {
+			if (model) {
+				const existingContent = model.getValue();
+				const delimiterIndex = existingContent.lastIndexOf(VSCODE_LSP_TERMINAL_PROMPT_TRACKER);
+
+				// Keep content only up to delimiter
+				const sanitizedExistingContent = delimiterIndex !== -1 ?
+					existingContent.substring(0, delimiterIndex) :
+					existingContent;
+
+				// Combine base content with new content
+				const newContent = sanitizedExistingContent + VSCODE_LSP_TERMINAL_PROMPT_TRACKER + content;
+
+				model.setValue(newContent);
+			}
+		}
+	}
+
+	private _registerTerminalCommandFinishedListener(): void {
+		const attachListener = () => {
+			if (this._onCommandFinishedListener.value) {
+				return;
+			}
+
+			// Inconsistent repro: Covering case where commandDetection is available but onCommandFinished becomes available later
+			if (this._commandDetection && this._commandDetection.onCommandFinished) {
+				this._onCommandFinishedListener.value = this._register(this._commandDetection.onCommandFinished((e) => {
+					if (e.exitCode === 0 && this._shellType === GeneralShellType.Python) {
+						this.setContent(e.command);
+					}
+
+				}));
+			}
+		};
+		attachListener();
+
+		// Listen to onDidAddCapabilityType because command detection is not available until later
+		this._register(this._capabilitiesStore.onDidAddCapabilityType(e => {
+			if (e === TerminalCapability.CommandDetection) {
+				this._commandDetection = this._capabilitiesStore.get(TerminalCapability.CommandDetection);
+				attachListener();
+			}
+		}));
+
+	}
+
+	// TODO: Adapt to support non-python virtual document for non-python REPLs.
+	async provideTextContent(resource: URI): Promise<ITextModel | null> {
+		const existing = this._modelService.getModel(resource);
+
+		if (existing && !existing.isDisposed()) {
+			return existing;
+		}
+
+		const extension = resource.path.split('.').pop();
+		let languageId: string | undefined | null = undefined;
+		if (extension) {
+			languageId = this._languageService.getLanguageIdByLanguageName(extension);
+
+			if (!languageId) {
+				switch (extension) {
+					case 'py': languageId = 'python'; break;
+					// case 'ps1': languageId = 'powershell'; break;
+					// case 'js': languageId = 'javascript'; break;
+					// case 'ts': languageId = 'typescript'; break; etc...
+				}
+			}
+		}
+
+		const languageSelection = languageId ?
+			this._languageService.createById(languageId) :
+			this._languageService.createById('plaintext');
+
+		return this._modelService.createModel('', languageSelection, resource, false);
+	}
+
+}
+
+/**
+ * Creates a terminal language virtual URI.
+ */
+// TODO: Make sure path is correct for Windows when we start supporting LSP completions in Windows. (Both Python, Non-Python)
+export function createTerminalLanguageVirtualUri(terminalId: number, languageExtension: string): URI {
```

Comment:
Left a TODO here so that when we make this more generic (not Python specific) in the future, we use proper / or \ depending on the different OS.

Reply:
Might want to make an issue so that we keep track of this TODO, including a hyperlink to it 😃

Reply:
Good idea, its right here now: [2c306b4](https://github.com/microsoft/vscode/pull/243528/commits/2c306b4ef1725c8a16774e37d0dcc128fd572784)